scriptworker/task.py

```diff
@@ -377,21 +364,30 @@ def extract_github_repo_owner_and_name(repo_url):
     if not repo_url.startswith('https://github.com/'):
         raise ValueError('{} is not a supported GitHub URL!'.format(repo_url))
 
-    if repo_url.endswith(".git"):
-        repo_url = repo_url[:-len(".git")]
-    elements = repo_url.split('/')
+    parts = _get_parts_of_url_path(repo_url)
+    repo_owner = parts[0]
+    repo_name = parts[1]
 
-    return elements[3], elements[4]
+    if repo_name.endswith(".git"):
+        repo_name = repo_name[:-len(".git")]
 
+    return repo_owner, repo_name
 
-# is_try {{{1
-def _is_try_url(url):
+
+def _get_parts_of_url_path(url):
     parsed = urlparse(url)
     path = unquote(parsed.path).lstrip('/')
     parts = path.split('/')
-    if "try" in parts[0]:
-        return True
-    return False
+    return parts
+
+
+def _is_try_url(url):
+    return 'try' in _get_parts_of_url_path(url)[0]
+
+
+def _does_url_come_from_official_repo(context, url):
+    official_repo_owner = context.config['official_github_repos_owner']
+    return not official_repo_owner or official_repo_owner == _get_parts_of_url_path(url)[0]
```

Comment:
Hm. Do we really want this to be `True` for non-mobile? I suppose we're checking this elsewhere. I might lean towards `True` / `False` / `None` for no owner specified. Not a blocker, just curious.

Reply:
Hmmm, I could use `None` and check for it later. I'll address this in a followup